multus_sriov_cni_install.sh

```diff
@@ -0,0 +1,202 @@
+export CNI_CONF_DIR=${CNI_CONF_DIR:-/etc/cni/net.d/}
+export ALLOW_PRIVILEGED=${ALLOW_PRIVILEGED:-true}
+export NET_PLUGIN=${NET_PLUGIN:-cni}
+
+export KUBE_ENABLE_CLUSTER_DNS=${KUBE_ENABLE_CLUSTER_DNS:-false}
+export API_HOST=$(hostname).$(hostname -y)
+export API_HOST_IP=$(hostname -I | awk '{print $1}')
+export KUBECONFIG=${KUBECONFIG:-/var/run/kubernetes/admin.kubeconfig}
+
+#TODO add autodiscovering
+export MACVLAN_INTERFACE=${MACVLAN_INTERFACE:-eno1}
+export SRIOV_INTERFACE=${SRIOV_INTERFACE:-enp3s0f0}
+export VFS_NUM=${VFS_NUM:-4}
+
+echo "Working under $WORKSPACE folder"
+mkdir -p $WORKSPACE
+mkdir -p $LOGDIR
+mkdir -p $ARTIFACTS
+mkdir -p $CNI_CONF_DIR
+mkdir -p $CNI_BIN_DIR
+
+pushd $WORKSPACE
+
+
+function configure_multus {
+
+    echo "Configure Multus"
+    kubectl create -f $WORKSPACE/multus-cni/images/multus-daemonset.yml
+    sleep 60
+    cat > $CNI_CONF_DIR/00-multus.conf <<EOF
+{
+    "name": "multus-cni-network",
+    "type": "multus",
+    "capabilities": {
+        "portMappings": true
+    },
+    "delegates": [
+        {
+            "cniVersion": "0.3.0",
+            "name": "macvlan-network",
+            "type": "macvlan",
+            "master": "$MACVLAN_INTERFACE",
+            "mode": "bridge",
+            "ipam": {
+                "type": "host-local",
+                "subnet": "192.168.1.0/24",
+                "rangeStart": "192.168.1.100",
+                "rangeEnd": "192.168.1.216",
+                "routes": [{"dst": "0.0.0.0/0"}],
+                "gateway": "192.168.1.1"
+            }
+        }
+    ],
+    "logFile": "$LOGDIR/multus.log",
+    "logLevel": "debug",
+    "kubeconfig": "$KUBECONFIG"
+}
+EOF
+    cp $CNI_CONF_DIR/00-multus.conf $ARTIFACTS
+    return $?
+}
+
+
+function download_cni {
+    if ! $RECLONE ; then
+        return 0
+    fi
+    #trap read debug
+
+    echo "Download $MULTUS_CNI_REPO"
+    git clone $MULTUS_CNI_REPO $WORKSPACE/multus-cni
+    cd $WORKSPACE/multus-cni
+    git checkout $MULTUS_CNI_BRANCH
+    git log -p -1 > $ARTIFACTS/multus-cni.txt
+    cd -
+
+    echo "Download $SRIOV_CNI_REPO"
+    git clone $SRIOV_CNI_REPO $WORKSPACE/sriov-cni
+    pushd $WORKSPACE/sriov-cni
+    git checkout $SRIOV_CNI_BRANCH
+    git log -p -1 > $ARTIFACTS/sriov-cni.txt
+    popd
+
+    echo "Download $PLUGINS_REPO"
+    git clone $PLUGINS_REPO $WORKSPACE/plugins
+    pushd $WORKSPACE/plugins
+    git checkout $PLUGINS_BRANCH
+    git log -p -1 > $ARTIFACTS/plugins.txt
+    bash ./build_linux.sh
+    cp $WORKSPACE/plugins/bin/* $CNI_BIN_DIR/
+    popd
+
+    echo "Download $SRIOV_NETWORK_DEVICE_PLUGIN_REPO"
+    git clone $SRIOV_NETWORK_DEVICE_PLUGIN_REPO $WORKSPACE/sriov-network-device-plugin
+    pushd $WORKSPACE/sriov-network-device-plugin
+    git checkout $SRIOV_NETWORK_DEVICE_PLUGIN_BRANCH
+    git log -p -1 > $ARTIFACTS/sriov-network-device-plugin.txt
+    make build
+    cp $WORKSPACE/sriov-network-device-plugin/build/sriovdp $CNI_BIN_DIR
```

Comment:
No need to copy `sriovdp` into `$CNI_BIN_DIR`, it's not a cni binary, will be executed explicitly.